### PR TITLE
*: fix windows tests

### LINKED ISSUE #2269: Fix logging on Windows
**Problem**
Absolute logger paths are not supported by zap logger (https://github.com/uber-go/zap/issues/621, https://github.com/uber-go/zap/pull/624). As example, see the failing test:
```
2021-11-17T15:46:00.8922873Z === RUN   TestHandleLoggingParams/default
2021-11-17T15:46:00.8923506Z     server_test.go:53: 
2021-11-17T15:46:00.8923992Z         	Error Trace:	server_test.go:53
2021-11-17T15:46:00.8924596Z         	Error:      	Received unexpected error:
2021-11-17T15:46:00.8925814Z         	            	couldn't open sink "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestHandleLoggingParams150626778\\001\\file.log": no sink found for scheme "c"
2021-11-17T15:46:00.8927627Z         	Test:       	TestHandleLoggingParams/default
2021-11-17T15:46:00.8928441Z === RUN   TestHandleLoggingParams/debug
2021-11-17T15:46:00.8929062Z     server_test.go:66: 
2021-11-17T15:46:00.8929637Z         	Error Trace:	server_test.go:66
2021-11-17T15:46:00.8930234Z         	Error:      	Received unexpected error:
2021-11-17T15:46:00.8931425Z         	            	couldn't open sink "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestHandleLoggingParams150626778\\001\\file.log": no sink found for scheme "c"
2021-11-17T15:46:00.8932684Z         	Test:       	TestHandleLoggingParams/debug
2021-11-17T15:46:00.8933483Z --- FAIL: TestHandleLoggingParams (0.00s)
2021-11-17T15:46:00.8934287Z     --- FAIL: TestHandleLoggingParams/default (0.00s)
2021-11-17T15:46:00.8935271Z     --- FAIL: TestHandleLoggingParams/debug (0.00s)
```

### LINKED ISSUE #2277: .github: adjust `Push images to DockerHub` workflow after Windows tests are fixed
Currently `publish_wsc` job is always run after `tests_wsc` job even if the tests are failing. After #2268 and #2269 Windows-based tests will be repaired, so we need to remove `${{ always() }}` condition from `publish_wsc` job.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #2269, close #2277.

Part of #2268 is also fixed, but dealing with `NEO-GO-VM > ←[J←[2K` is still in progress.

### CODE REVIEW

File: cli/server/server.go [hunk 1]
```diff
@@ -148,9 +153,8 @@ func handleLoggingParams(ctx *cli.Context, cfg config.ApplicationConfiguration)
 
 	if logPath := cfg.LogPath; logPath != "" {
 		if err := io.MakeDirForFile(logPath, "logger"); err != nil {
-			return nil, err
+			return nil, nil, err
 		}
-
```
Comment (roman-khimov) [inline cli/server/server.go:147]: Meh.

File: cli/server/server.go [hunk 2]
```diff
@@ -262,10 +277,15 @@ func restoreDB(ctx *cli.Context) error {
 	if err != nil {
 		return err
 	}
-	log, err := handleLoggingParams(ctx, cfg.ApplicationConfiguration)
+	log, logCloser, err := handleLoggingParams(ctx, cfg.ApplicationConfiguration)
 	if err != nil {
 		return cli.NewExitError(err, 1)
 	}
+	defer func() {
+		if logCloser != nil {
```
Comment (roman-khimov) [inline cli/server/server.go:285]: `if logCloser != nil { defer logCloser() }`?

File: cli/server/server.go [hunk 3]
```diff
@@ -145,6 +151,43 @@ func handleLoggingParams(ctx *cli.Context, cfg config.ApplicationConfiguration)
...
+						return nil, fmt.Errorf("file URLs must leave host empty or use localhost: got %v", u)
+					}
+					switch u.Path {
+					case "stdout":
+						return os.Stdout, nil
+					case "stderr":
+						return os.Stderr, nil
+					}
+					return os.OpenFile(u.Path[1:], // Remove leading slash left after url.Parse.
+						os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0666)
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply cli/server/server.go:185]: It's 0666 in the zap library and for linux, see https://github.com/uber-go/zap/pull/862 and https://github.com/uber-go/zap/blob/6f34060764b5ea1367eecda380ba8a9a0de3f0e6/sink.go#L139.

File: pkg/vm/cli/cli_test.go [hunk 4]
```diff
@@ -318,48 +318,54 @@ func TestRunWithDifferentArguments(t *testing.T) {
...
-		"run _ 1 2",
-		"loadbase64 "+base64.StdEncoding.EncodeToString([]byte{byte(opcode.MUL)}),
-		"run _ 21 2",
-	)
-
+	e.runProg(t, "loadgo "+filename, "run notexists")
 	e.checkNextLine(t, "READY: loaded \\d.* instructions")
 	e.checkNextLine(t, "Error:")
 
+	e = newTestVMCLI(t)
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/vm/cli/cli_test.go:325]: Can we raise it just for this particular test? Running a set of processes is more overhead than running one process.

File: cli/server_test.go [hunk 5]
```diff
@@ -96,29 +97,33 @@ func TestServerStart(t *testing.T) {
...
-		var line string
-		require.Eventually(t, func() bool {
-			line, err = e.Out.ReadString('\n')
-			if err != nil && err != io.EOF {
-				t.Fatalf(fmt.Sprintf("unexpected error while reading CLI output: %s", err))
+			var line string
+			require.Eventually(t, func() bool {
+				line, err = e.Out.ReadString('\n')
+				if err != nil && err != io.EOF {
+					t.Fatalf(fmt.Sprintf("unexpected error while reading CLI output: %s", err))
```
Comment (fyrchik) [inline cli/server_test.go:114]: No need to use `fmt.Sprintf` with `Fatalf`.

File: pkg/vm/cli/cli_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/vm/cli/cli_test.go:325]: Or we can set it base on the size of `commands` slice, like `len(commands) * time.Second / 2`.

File: cli/server/server.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (roman-khimov) [reply cli/server/server.go:186]: Does it work with 0644? 0666 makes me feel uncomfortable.

File: pkg/vm/cli/cli_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/vm/cli/cli_test.go]: Fixed with new executor method with a customisable timeout.

File: pkg/compiler/analysis.go [hunk 6]
```diff
@@ -235,10 +234,6 @@ func (c *codegen) fillDocumentInfo() {
 	fset := c.buildInfo.config.Fset
 	fset.Iterate(func(f *token.File) bool {
 		filePath := f.Position(f.Pos(0)).Filename
-		filePath, err := filepath.Rel(c.buildInfo.config.Dir, filePath)
```
Comment (roman-khimov) [inline pkg/compiler/analysis.go:238]: Usually, absolute paths in debug files are to be avoided. Maybe we can distinguish between contract files (that shouldn't have this problem) and interops?

Comment (AnnaShaleva) [reply pkg/compiler/analysis.go:238]: @fyrchik told me about that also, but I wasn't sure whether it's OK to combine both relative and absolute file formats.

Fixed, could you check?

Comment (AnnaShaleva) [reply pkg/compiler/analysis.go:238]: Oh, not fixed yet.

Comment (roman-khimov) [reply pkg/compiler/analysis.go:238]: Or maybe always store absolute path if relative can't be derived (`err` handling essentially)?

Comment (AnnaShaleva) [reply pkg/compiler/analysis.go:238]: Yes, I like this solution the most, implemented.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**